main.cpp

```diff
@@ -331,160 +286,58 @@ DoubleType& DoubleType::divide( double val)
     {
         std::cout << "warning: floating point division by zero!" << "\n";
     }
-    *pt /= val;
+    else if(pt != nullptr)
+    {
+        *pt /= val;    
+    }
+    
```

Comment:
floating point division by zero is a warning, not an error.  you still need to do the division.
- [ ] revise